splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/index/IndexTransformer.java

```diff
@@ -477,11 +483,13 @@ public boolean areIndexKeysModified(KVPair mutation, BitSet indexedColumns) {
         EntryDecoder newPutDecoder = new EntryDecoder();
         newPutDecoder.set(mutation.getValue());
         BitIndex updateIndex = newPutDecoder.getCurrentIndex();
-        for (int i = updateIndex.nextSetBit(0); i >= 0; i = updateIndex.nextSetBit(i + 1)) {
-            if (indexedColumns.get(i))
-                return true;
+        BitSetIterator iterator = nonPkIndexedCols.iterator();
+        int nextBit;
+        while((nextBit = iterator.nextSetBit()) != -1) {
+            if (!updateIndex.isSet(nextBit))
+                return false;
         }
-        return false;
+        return true;
```

Comment:
It looks like parameter `indexedColumns` is not needed after the change.

Reply:
Good catch, removed.